### DISCUSSION TITLE
How can i create such a schema?

### DISCUSSION THREAD

**Discussion by halotest** [prisma/prisma#9609] [Q&A] [answered]
Hello! I'm struggling with proper schema creation.  I have one parent model "Product" and a few child models(Product-Type-A, Product-Type-B, Product-Type-C) each of which represent a different product type. Basically I want parent (Product) to be created when i create a new Type-* object and its (Parent) id field to point to children object. The problem i'm also facing is that parent(Product) model id can refer to different child models, it can be either Type-A, Type-B or Type-C. Also it's a 1-1 relation so when product-type-*  is created, one Product is created too and refers to children.
In the end i want to be able to retrieve all of the products by querying parent Product model like prisma.product.findMany()

So far my schema looks like this: 

```prisma
model Product {
   id             String              @id @default(dbgenerated()) @map("_id") @db.ObjectId
   typeA          Product-TypeA?      @relation(fields: [productId], references: [id])   
   typeB          Product-TypeB?      @relation(fields: [productId], references: [id])   
   productId      String?             @db.ObjectId      // Can be either a typeA or typeB
}

```

```prisma
model Product-TypeA {
    id        String    @id @default(dbgenerated()) @map("_id") @db.ObjectId
    product   Product[]
    ....
}

```
```prisma
model  Product-TypeB {
    id        String    @id @default(dbgenerated()) @map("_id") @db.ObjectId
    product   Product[]
    ....
}

```

And with this schema i'm getting such error: Unknown arg `product` in data.product
but my VSCode autocompletion clearly tells me data.product.create is valid.

Thanks in advance!

**Comment by ryands17** [comment, marked as answer]
@halotest 👋 

Unfortunately we don't support Unions at the moment, so you need to add separate models (separate `productId`) for `typeA` and `typeB`. We have a feature request for this so it would be great if you could add a 👍 so that we can know the priority for this 🙂 
https://github.com/prisma/prisma/issues/8838

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
